nomad/mock/mock.go

```diff
@@ -1286,3 +1286,40 @@ func JobWithScalingPolicy() (*structs.Job, *structs.ScalingPolicy) {
 	job.TaskGroups[0].Scaling = policy
 	return job, policy
 }
+
```

Comment:
There's not a lot of places we can use these mocks b/c most of our tests need the nodes as well, but it's a nice-to-have for future work or test refactoring.